lib/vnet/user_process_windows.go

```diff
@@ -42,7 +46,26 @@ func runPlatformUserProcess(ctx context.Context, config *UserProcessConfig) (pm
 		}
 	}()
 
-	listener, err := net.Listen("tcp", ":0")
+	ipcCreds, err := newIPCCredentials()
+	if err != nil {
+		return nil, trace.Wrap(err, "creating credentials for IPC")
+	}
+	serverTLSConfig, err := ipcCreds.server.serverTLSConfig()
+	if err != nil {
+		return nil, trace.Wrap(err, "generating gRPC server TLS config")
+	}
+	credDir, err := os.MkdirTemp("", "vnet_service_certs")
+	if err != nil {
+		return nil, trace.Wrap(err, "creating temp dir for service certs")
+	}
+	if err := secureCredDir(credDir); err != nil {
+		return nil, trace.Wrap(err, "applying permissions to service credential dir")
+	}
+	if err := ipcCreds.client.write(credDir); err != nil {
+		return nil, trace.Wrap(err, "writing service IPC credentials")
+	}
+
+	listener, err := net.Listen("tcp", "localhost:0")
 	if err != nil {
 		return nil, trace.Wrap(err, "listening on tcp socket")
 	}
```

Comment:
```suggestion
	}
	// The grpcServer below takes ownership of listener.
```

Reply:
thanks, this prompted me to call GracefulStop on the grpc server instead of closing the listener

Reply:
Great!